jupyter_server_ydoc/handlers.py

```diff
@@ -209,14 +225,26 @@ async def recv(self):
         return message
 
     def on_message(self, message):
+        assert self.websocket_server is not None
         if message[0] == AWARENESS:
             # awareness
             skip = False
-            # changes = self.room.awareness.get_changes(msg)
+            changes = self.room.awareness.get_changes(message[1:])
+            added_users = changes["added"]
+            removed_users = changes["removed"]
+            for i, user in enumerate(added_users):
+                name = changes["states"][i]["user"]["name"]
```

Comment:
When I test this I get a KeyError on `user` here. Thoughts?

Reply:
The [awareness decoding in ypy-websocket](https://github.com/y-crdt/ypy-websocket/blob/9fb2fad3a20d8df833f8ea462d2522847d435ecf/ypy_websocket/awareness.py) was just me trying to reverse-engineer the [Yjs awareness protocol](https://github.com/yjs/y-protocols/blob/255b7e6c96d9c6dd48b5519b3f83ab65afdb4bba/awareness.js), but it might be incomplete.
I'll just test for the presence of the `"user"` key for now. Maybe @dmonad knows more?